include/caffe/util/db.hpp

```diff
@@ -181,6 +181,87 @@ class LMDB : public DB {
   MDB_dbi mdb_dbi_;
 };
 
+
+#define MAX_BUF 10485760  // max entry size
+class DatumFileCursor : public Cursor {
+ public:
+  explicit DatumFileCursor(const string& path) {
+    this->path = path;
+    in = NULL;
+    SeekToFirst();
+  }
+  virtual ~DatumFileCursor() {
+  }
+  virtual void SeekToFirst();
+
+  virtual void Next();
+
+  virtual string key() {
```

Comment:
It should fail instead of giving a warning. Ex:
CHECK(valid()) << "not valid state";